addons/web_editor/controllers/main.py

```diff
@@ -270,130 +270,182 @@ def get_image_info(self, image_id=None, xml_id=None):
     ## @param key - the xml_id or id of the view the resources are related to
     ## @param get_views - True if the views must be fetched (default to True)
     ## @param get_scss - True if the style must be fetched (default to True)
+    ## @param get_js - True if the javascript must be fetched (default to True)
     ## @param bundles - True if the bundles views must be fetched (default to False)
     ## @param bundles_restriction - Names of the bundle in which to look for scss files (if empty, search in all of them)
+    ## @param only_user_custom_files - True if only user custom files are needed (default to True)
     ## @returns a dictionary with views info in the views key and style info in the scss key
     @http.route("/web_editor/get_assets_editor_resources", type="json", auth="user", website=True)
-    def get_assets_editor_resources(self, key, get_views=True, get_scss=True, bundles=False, bundles_restriction=[]):
+    def get_assets_editor_resources(self, key, get_views=True, get_scss=True, get_js=True, bundles=False, bundles_restriction=[], only_user_custom_files=True):
         # Related views must be fetched if the user wants the views and/or the style
         views = request.env["ir.ui.view"].get_related_views(key, bundles=bundles)
         views = views.read(['name', 'id', 'key', 'xml_id', 'arch', 'active', 'inherit_id'])
 
-        scss_files_data_by_bundle = []
-
-        # Load scss only if asked by the user
+        if get_js:
+            js_files_data_by_bundle = self.get_resources('js', views, only_user_custom_files, bundles_restriction)
```

Comment:
For consistency with the `get_assets_editor_resources` function I would put `bundles_restriction` param before `only_user_custom_files` param